### ISSUE TITLE
widget randomly dies

### ISSUE THREAD

**Issue by Youdaman** [ncsoft/Unreal.js#56]
I have the following code:

Near the start of my game.js
```
  //let widgetClass = Blueprint.Load('/Game/HUDWidget').GeneratedClass 
  let widgetClass = Root.ResolveClass('HUDWidget') 
```

In my PlayerController class
```
    ReceiveBeginPlay() {
      this.widget = GWorld.CreateWidget(widgetClass, this)
      if (!this.widget.IsInViewport()) this.widget.AddToViewport()
      //global.widget = this.widget
    }
```

In the ReceiveTick function of my PlayerController
```
      this.widget.SprintBar.SetPercent(this.sprintTimeLeft / this.sprintTime)
```

That last line above throws errors after I play my game for a little while, see near end of this video:

https://www.youtube.com/watch?v=Fl8zW9CxPF8

Basically the widget becomes undefined. Any idea why? Do widgets have a lifespan or heartbeat that you need to keep calling in order to keep them "alive"?

**Comment by Youdaman** [comment]
I think I've worked out the problem but I'm still troubleshooting -- I added some debug code to see what the values for the variables were at the moment `this.widget` becomes undefined, and it happens when `this.sprintTimeLeft` and `this.sprintTime` are undefined.

So that makes sense in that I'm attempting `.SetPercent( undefined / undefined)` which causes the widget to crash. I'll reopen this if it turns out to be something related to Unreal.js but it looks like it's just my logic causing those variables to be undefined.
